base_layer/key_manager/src/lib.rs

```diff
@@ -30,3 +38,31 @@ pub(crate) fn mac_domain_hasher<D: Digest + LengthExtensionAttackResistant>(
 ) -> DomainSeparatedHasher<D, KeyManagerDomain> {
     DomainSeparatedHasher::<D, KeyManagerDomain>::new_with_label(label)
 }
+
+hidden_type!(CipherSeedEncryptionKey, [u8; CIPHER_SEED_ENCRYPTION_KEY_BYTES]);
+hidden_type!(CipherSeedMacKey, [u8; CIPHER_SEED_MAC_KEY_BYTES]);
+
+#[derive(Debug, Clone)]
+pub struct SeedWords {
+    words: Vec<Hidden<String>>,
+}
+
+impl SeedWords {
+    pub fn new(words: &[String]) -> Self {
+        Self {
+            words: words.into_iter().map(|m| Hidden::hide(m.clone())).collect::<Vec<_>>(),
```

Comment:
Clone may create extra copies in the db, this should work
```suggestion
            words: words.into_iter().map(Hidden::hide).collect::<Vec<_>>(),
```

Reply:
Agree. You could also zeroize the strings in the vec that you pass in

Reply:
Thanks for the suggestions, I refactored the input type to `Vec<Hidden<String>>`.